uPortal-api/uPortal-api-rest/src/main/java/org/apereo/portal/json/rendering/JsonWrapperFilteringCharacterPipelineComponent.java

```diff
@@ -42,13 +45,17 @@ public CacheKey getCacheKey(HttpServletRequest request, HttpServletResponse resp
     @Override
     public PipelineEventReader<CharacterEventReader, CharacterEvent> getEventReader(
             HttpServletRequest request, HttpServletResponse response) {
-        final PipelineEventReader<CharacterEventReader, CharacterEvent> pipelineEventReader =
-                this.wrappedComponent.getEventReader(request, response);
-        final CharacterEventReader eventReader = pipelineEventReader.getEventReader();
-        final JsonWrapperFilteringCharacterEventReader jsonWrapperFilteringCharacterEventReader =
-                new JsonWrapperFilteringCharacterEventReader(eventReader);
-        final Map<String, String> outputProperties = pipelineEventReader.getOutputProperties();
-        return new PipelineEventReaderImpl<CharacterEventReader, CharacterEvent>(
-                jsonWrapperFilteringCharacterEventReader, outputProperties);
+        if (this.wrappedComponent != null) {
+            final PipelineEventReader<CharacterEventReader, CharacterEvent> pipelineEventReader =
+                    this.wrappedComponent.getEventReader(request, response);
+            final CharacterEventReader eventReader = pipelineEventReader.getEventReader();
+            final JsonWrapperFilteringCharacterEventReader
+                    jsonWrapperFilteringCharacterEventReader =
+                            new JsonWrapperFilteringCharacterEventReader(eventReader);
+            final Map<String, String> outputProperties = pipelineEventReader.getOutputProperties();
+            return new PipelineEventReaderImpl<CharacterEventReader, CharacterEvent>(
+                    jsonWrapperFilteringCharacterEventReader, outputProperties);
+        }
+        return null;
     }
```

Comment:
wrappedComponent is a field from PipelineComponentWrapper. You can either throw IllegalStateException or return null and let caller method handle the null value. We can go for IllegalStateException.